### PR TITLE
RFC: Observability documentation reorg

### PR DESCRIPTION
**Pull request by MS-Santi**
First pass at the observability documentation.
Super rough, but opening it early for comment.

Notice there is a placeholder for Hugo's content to come.

### CODE REVIEW

File: observabilityV1/guidance/eventLogging.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+- Unrecoverable errors from startup.
+- Warnings if application still runnable, but not as expected (i.e. not providing blob connection string, thus resorting to local files. Another example is if there's a need to fail back to a secondary service or a known good state, because it didn’t get an answer from a primary dependency.
+- Information about the service’s state at startup (build #, configs loaded, etc.)
+
+**Per incoming request**:
+
+- Basic information for each incoming request: the url (scrubbed of any personally identifying data, a.k.a. PII), any user/tenant/request dimensions, response code returned, request-to-response latency, payload size, record counts, etc. (whatever you need to learn something from the aggregate data)
+- Warning for any unexpected exceptions, caught only at the top controller/interceptor and logged with or alongside the request info, with stack trace. Return a 500. This code doesn’t know what happened.
+
```
Comment (ViniciusSouza) [inline observabilityV1/guidance/eventLogging.md]: I would mention at this point as well the usage of correlation ID, thinking on microservices and even a easy way to correlate the logs within the service from a trigger point (request, event, etc.)

We got one sample of that when the service does use a correlation header (used by NewRelic), but the log does not show this information, when the agent scraps the logs it is impossible to correlate those.

File: observabilityV1/guidance/readme.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# General Guidance
```
Comment (ViniciusSouza) [inline observabilityV1/guidance/readme.md]: Loved this section 😀

File: observabilityV1/guidance/eventLogging.md (hunk 1 above)
Comment (ViniciusSouza) [reply observabilityV1/guidance/eventLogging.md]: Just saw the information on Tracing.md 😁

**Review by ViniciusSouza — APPROVED** [review]
I really like the changes

**Comment by MS-Santi** [comment]
Abandoning to implement the new structure as discussed with the team

**Pull request closed without merging**